src/vs/platform/terminal/common/xterm/shellIntegrationAddon.ts

```diff
@@ -226,6 +226,13 @@ const enum VSCodeOscPt {
 	 */
 	SetMark = 'SetMark',
 
+	/**
+	 * Clears any remaining environment variables in the cached environment
+	 *
+	 * Format: `OSC 633 ; EnvClear ; <Nonce>`
+	 */
+	EnvClear = 'EnvClear',
```

Comment:
I like the start/end sequences as we can make that transactional if we need to; we would know if there is access to an environment variable that may be getting updated. Could we instead add a clear flag to `EnvSingleStart` to make it: 

```
Format: `OSC 633 ; EnvSingleStart [; <Clear>] ; <Nonce>`
```

Where clear is an described in the comment as 1 or 0?